src/api/api.ts

```diff
@@ -0,0 +1,185 @@
+import { ethers, type Wallet, type Contract } from 'ethers'
+import { type ExtensionContext } from 'vscode'
+import { extractPvtKey, listAddresses } from '../utils/wallet'
+import * as vscode from 'vscode'
+import {
+  getNetworkNames,
+  getSelectedNetConf,
+  getSelectedProvider
+} from '../utils/networks'
+import { type ContractABI, type CompiledJSONOutput, type NetworkConfig } from '../types'
+import {
+  getConstructorInputFullPath,
+  getDeployedFullPath,
+  getFunctionInputFullPath
+} from '../utils/functions'
+import { type JsonFragment } from '@ethersproject/abi'
+
+const event: {
+  network: vscode.EventEmitter<string>
+  account: vscode.EventEmitter<string>
+} = {
+  network: new vscode.EventEmitter<string>(),
+  account: new vscode.EventEmitter<string>()
+}
+
+// PROVIDER
+const providerDefault = (context: ExtensionContext): any => {
+  return getSelectedProvider(context)
+}
+
+const getAvailableNetwork = (): string[] => {
+  return getNetworkNames()
+}
+
+const getNetwork = (context: ExtensionContext): NetworkConfig => {
+  return getSelectedNetConf(context)
+}
+
+const setNetwork = (context: ExtensionContext, network: string): string => {
+  if (network === null) {
+    return 'Network parameter not given'
+  }
+  if (!getNetworkNames().includes(network)) {
+    return 'Network not found'
+  } else {
+    void context.workspaceState.update('selectedNetwork', network)
+    return 'Network changed to ' + network
+  }
+}
+
+// WALLETS
+const getWallet = async (context: ExtensionContext, account: string): Promise<Wallet> => {
+  const address: any = await context.workspaceState.get('account')
+  account = account ?? address
+  const provider = getSelectedProvider(context)
+  const privateKey = await extractPvtKey(context.extensionPath, account)
+  const wallet = new ethers.Wallet(privateKey, provider)
+  return wallet
+}
+
+const listAllWallet = async (context: ExtensionContext): Promise<string[]> => {
+  const result = await listAddresses(context, context.extensionPath)
+  return result
+}
+
+// CONTRACT
+
+const getContract = async (
+  context: ExtensionContext,
+  address: string,
+  abi: any,
+  wallet: ethers.Signer
+): Promise<Contract> => {
+  const contract = new ethers.Contract(address, abi, wallet)
+  return contract
+}
+
+const listFunctions = (abi: any): any => {
+  const result = []
```

Comment:
try to use ` const functionsAbi = abi.filter(
    (i) => i.type === 'function'
  )`

Reply:
this function is not used we removed this issue